### PR TITLE
Replace ps | grep with a simple pgrep

### PR DESCRIPTION
**Pull request by neheb**
Eliminates busybox hacks and allows it to work properly with busybox.

### CODE REVIEW

File: scripts/airmon-ng.linux [hunk 1]
```diff
@@ -1120,12 +1104,7 @@ scanProcesses() {
 	fi
 
 	unset match
-	if [ "${pso}" = 1 ]; then
-		match="$(ps ${psopts} -o comm= | grep ${PROCESSES} | grep -v grep | wc -l)"
-	elif [ "${pso}" = 0 ]; then
-		#openwrt busybox grep hits on itself so we -v it out
-		match="$(ps ${psopts} | grep ${PROCESSES} | grep -v grep | wc -l)"
-	fi
+	match="$(pgrep ${PROCESSES} | grep -c -v grep)"
```
Comment (neheb) [inline scripts/airmon-ng.linux]: Actually the grep here seems like it can be replaced by wc -l.

**Comment by aircrack-ng** [comment]
Doesn't work on Kali (and probably other distros that don't use busybox). Output is empty with a patched airmon-ng, and reports processes on unpatched.

**Comment by jbenden** [comment]
I may be wrong, but I believe that `pgrep` is not available across the wide range of supported operating systems. That said, for Debian-based distros, the `pgrep` tool is provided by the `procps` package.

**Comment by neheb** [comment]
I see the problem... pgrep can only handle one expression at a time. It does not support the \| syntax used in PROCESSES . Meaning a loop would need to be created...

I doubt that would simplify the script. Closing.

**Pull request closed without merging**